src/core/net/udp6.cpp

```diff
@@ -156,7 +172,10 @@ otError Udp::RemoveReceiver(Receiver &aReceiver)
 
 otError Udp::Open(SocketHandle &aSocket, otUdpReceive aHandler, void *aContext)
 {
-    otError error = OT_ERROR_NONE;
+    otError       error = OT_ERROR_NONE;
+    SocketHandle *prev;
+
+    VerifyOrExit(mSockets.Find(aSocket, prev) == OT_ERROR_NOT_FOUND, error = OT_ERROR_ALREADY);
```

Comment:
Turn out `Udp::Open` is not **reentrant**. 
Currently `BackboneTmf` agent would call `Coap::Start` when BBR primary state changes, which in turn calls `Udp::Open`.
So I make `Udp::Open` **reentrant** as the code shows. 
@librasungirl  @abtink @bukepo